edweiss-app/app/(app)/courses/[id]/index.tsx

```diff
@@ -125,39 +119,54 @@ const CoursePage: ApplicationRoute = () => {
 		});
 
 	// Filter previous assignments
-	const previousAssignments = assignments.filter((assignment) => assignment.dueDate.seconds * timeInMS.SECOND <= new Date().getTime()).reverse();
+	const previousAssignments = useMemo(() => {
+		return assignments
+			.filter(
+				(assignment) =>
+					assignment.dueDate.seconds * timeInMS.SECOND <= new Date().getTime()
+			)
+			.reverse();
+	}, [assignments, timeInMS.SECOND]);
 
 	// Filter upcoming assignments
-	const upcomingAssignments = assignments.filter((assignment) => assignment.dueDate.seconds * timeInMS.SECOND > new Date().getTime());
+	const upcomingAssignments = useMemo(() => {
+		const filteredAssignments = assignments.filter(
+			(assignment) =>
+				assignment.dueDate.seconds * timeInMS.SECOND > new Date().getTime()
```

Comment:
This predicate seems to repeat often in this file, maybe it's the occasion to refactor it into a `utils/time.ts` function for better maintainability and readability?

Reply:
The predicate repeats 2 times and returns 2 different things, in addition they are too specific to assignments to be placed in `utils/time.ts` in my opinion. However, I implemented the logic of `isBeforeNow` and `isAfterNow` to compare a TimeStamp with the current time. Thank you for pointing this possible improvement of this part!